tests/reproducibility/test_graphsage.py

```diff
@@ -100,3 +158,14 @@ def test_reproducibility(petersen_graph, shuffle):
             shuffle=shuffle,
         )
     )
+
+
+@pytest.mark.parametrize("shuffle", [True, False])
+def test_nai(petersen_graph, shuffle):
+    target_size = 10
+    targets = np.random.rand(len(petersen_graph.nodes()), target_size)
+    assert_reproducible(
+        lambda: gs_nai(
+            petersen_graph, targets, [2], tf.optimizers.Adam(1e-3), shuffle=shuffle
```

Comment:
Is it worth checking that multiple layers are reproducible, to ensure the sampling is doing the right thing?

```suggestion
            petersen_graph, targets, [2, 2], tf.optimizers.Adam(1e-3), shuffle=shuffle
```

Reply:
good point!